### PR TITLE
Generate LRO helper methods in client class

### PR DESCRIPTION
**Pull request by chrisdunelm**
Methods that require generating in the client clas if there are any LRO methods in this client service.

Further codegen and test execution will be in a future PR.

### CODE REVIEW

**Comment by chrisdunelm** [comment]
The files in src/Google/LongRunning/... are generated from protoc, so can be mostly ignored.

File: src/Ast/AST.php [hunk 1]
```diff
@@ -461,6 +461,33 @@ public function toCode(): string
         };
     }
 
+    /**
+     * Create a ternary expression.
```
Comment (jskeet) [inline src/Ast/AST.php:465]: Well, creates a conditional ?: expression.
You know my feelings about using the arity of an expression as its name ;)

**Review by jskeet — APPROVED** [review]
Structure-only review, but looks basically fine.

File: src/Generation/MethodDetails.php [hunk 2]
```diff
@@ -39,8 +45,24 @@ private static function createLro(ServiceDetails $svc, MethodDescriptorProto $de
             public function __construct($svc, $desc)
             {
                 parent::__construct($svc, $desc);
-                // TODO: Load LRO-specific details.
+                $this->methodType = MethodDetails::LRO;
+                $this->isLro = true;
+                $catalog = $svc->catalog;
+                $lroData = ProtoHelpers::getCustomOption($desc, CustomOptions::GOOGLE_LONGRUNNING_OPERATIONINFO, OperationInfo::class);
+                if (is_null($lroData)) {
+                    throw new \Exception('An LRO method must provide an `google.api.operation` option.');
```
Comment (amanda-tarafa) [inline src/Generation/MethodDetails.php:53]: nit:
```suggestion
                    throw new \Exception('An LRO method must provide a `google.api.operation` option.');
```

File: tests/ProtoTests/BasicLro/Gapic/BasicLroGapicClient.php [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+     *
+     * @return OperationResponse
+     *
+     * @experimental
+     */
+    public function resumeOperation($operationName, $methodName = null)
+    {
+        $options = isset($this->descriptors[$methodName]['longRunning']) ? $this->descriptors[$methodName]['longRunning'] : [];
+        $operation = new OperationResponse($operationName, $this->getOperationsClient(), $options);
+        $operation->reload();
```
Comment (amanda-tarafa) [inline tests/ProtoTests/BasicLro/Gapic/BasicLroGapicClient.php:109]: Just to double check that reload doesn't return a new OperationResponse.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Review by miraleung — APPROVED** [review]

File: src/Ast/AST.php (hunk 1 above)
Comment (chrisdunelm) [reply src/Ast/AST.php]: Comment changed :)

File: src/Generation/MethodDetails.php (hunk 2 above)
Comment (chrisdunelm) [reply src/Generation/MethodDetails.php]: Done, thanks

File: tests/ProtoTests/BasicLro/Gapic/BasicLroGapicClient.php (hunk 3 above)
Comment (chrisdunelm) [reply tests/ProtoTests/BasicLro/Gapic/BasicLroGapicClient.php:109]: Thanks for checking - this code is correct in that `reload` does not return a new OperationResponse (unlike C#).
This line is modelled from the monolithic generator here: https://github.com/googleapis/gapic-generator/blob/ae9338ab74e0fd91ee2814b348d59562693a76a6/src/main/resources/com/google/api/codegen/php/client_impl.snip#L303

**Pull request merged**